paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/pool/CachedClientPool.java

```diff
@@ -239,11 +289,58 @@ public String value() {
             return value;
         }
 
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            ConfElement other = (ConfElement) obj;
+            return Objects.equals(key, other.key) && Objects.equals(value, other.value);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(key, value);
+        }
+
         public static ConfElement of(String key, String value) {
             return new ConfElement(key, value);
         }
     }
 
+    static class UserGroupInformationConf {
```

Comment:
Why need to introduce `UserGroupInformationConf`? Revert this?

Reply:
It seems that "UserGroupInformation" returns different object instances for the same user, which cannot be used as a cache key.
![image](https://github.com/user-attachments/assets/a01c61c9-d980-469a-8a73-8cd39d865176)